template_lib.py

```diff
@@ -3,10 +3,12 @@
 import shlex
 import subprocess
 import sys
+from typing import Any
+from typing import Type
 
 import markdown_code_blocks
 import markupsafe
-
+import mistune
```

Comment:
this is not what I had imagined -- it should be something much simpler like this:

```python
copyable_s = '  # copyable'
copyable = lang.endswith(copyable_s)
lang, _, _ = lang.rpartition(copyable_s)
...

return f'<div class="copyable">{ret}</div>'
```

Reply:
- The regex for parsing `lang` lives [here](https://github.com/lepture/mistune/blob/2837628008dbe1d8f01e76845fcb00c157e28667/mistune.py#L112), and in order to be able to catch ` # copyable`, I thought that this regex ought to be overridden, that led making custom `Lexer` and `Grammar` classes.
- I might be missing the spot where your example here works, can you please direct me there?

Reply:
`def block_code` -- you already edited it :)

Reply:
- also, I thought about adding additional `div` around `ret`, but then I switched to using the original `div` to keep the structure consistent 😄

Reply:
- sorry if I argue, but I test with this minimal sample: https://pastebin.com/10UaNLeY , and `lang` is never parsed if it does not match the regex defined [here](https://github.com/lepture/mistune/blob/2837628008dbe1d8f01e76845fcb00c157e28667/mistune.py#L112), and as a result the `block_code` function is not called 🤔

Reply:
then how about `rst#copyable` ?

Reply:
- `rst#copyable` sure works, I tool your [example](https://github.com/pre-commit/pre-commit.github.io/pull/331#discussion_r407777535) so literally 😄 
- I will re-do it to accept such `rst#copyable` lang

Reply:
- btw, it might be worth mentioning that the current implementation supports not only `copyable` class but also any other class, so if you needed to add a print button to specific code blocks, it will be as easy as adding `printable` class (for example), and react to it in a javscript code.
- and to make the story even funnier, I wanted to go further and support multiple classes i.e: ` # copyable printable .. etc` on the same line, but I thought I should get your feedback first, and glad I did 😆

Reply:
yagni though :)

Reply:
done!